etcdserver/server.go

```diff
@@ -323,9 +326,8 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		return nil, fmt.Errorf("unsupported bootstrap config")
 	}
 
-	err = os.MkdirAll(cfg.MemberDir(), privateDirMode)
-	if err != nil && err != os.ErrExist {
-		return nil, err
+	if mkerr := fileutil.MkdirAllAlways(cfg.MemberDir()); mkerr != nil {
```

Comment:
I think there's a bug here if there's no `IsDirWriteableTest` associated with the `MkdirAllAlways`